src/test/java/BaseTest.java

```diff
@@ -1,13 +1,132 @@
+
+       WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+       wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//a[@href='#!/playlist/26571']"))).click();
+    }
+    @BeforeMethod
+    @Parameters({"BaseUrl"})
+    public void launchBrowser() {
+        driver = new ChromeDriver();
+        wait = new WebDriverWait(driver, Duration.ofSeconds(4));
+        url = "https://bbb.testpro.io/";
+        driver.get(url);
+    }
     @BeforeSuite
     public static void chromeConfigs() {
         // This is for Windows users
         if (System.getProperty("os.name").toLowerCase().contains("win")) {
             System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
         }
     }
+    public static void clickSubmitBtn() {
+        WebElement loginButton= BaseTest.driver.findElement(By.cssSelector("button[type='submit']"));
+        loginButton.click();
+    }
+    public static void providePassword(String password) {
+        WebElement passwordField = BaseTest.driver.findElement(By.cssSelector("[type = 'password']"));
+        passwordField.click();
+        passwordField.sendKeys(password);
+    }
+    public static void provideEmail(String email) {
+        WebElement emailField = BaseTest.driver.findElement(By.cssSelector("[type = 'email']"));
+        emailField.click();
+        emailField.sendKeys(email);
+    }
+    protected static void validation() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("a[href*='playlist/24815']"))).click();
+        WebElement songWasAdded = driver.findElement(By.xpath("//tr[@class='song-item']//td[contains(text(),'Epic Song')]"));
+        Assert.assertTrue(songWasAdded.isDisplayed());
+    }
+    protected static void abdisPlaylist() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//section[@id='songsWrapper']//li[5]"))).click();
+    }
+    protected static void addSongToPlaylistBtn() throws InterruptedException {
 
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[@class='btn-add-to']"))).click();
+    }
+    protected static void addingNewSong() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//tr[@class='song-item']//td[contains(text(),'Epic Song')]"))).click();
+    }
+    protected static void allSongsTab() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//a[@class='songs']"))).click();
```

Comment:
You can make one method and then use it so as not to repeat yourself, for example

  ```
protected static WebElement waitForElementToBeClickable(WebElement element) {
    return new WebDriverWait(driver, Duration.ofSeconds(10))
        .until(ExpectedConditions.elementToBeClickable(element));
  }
```

![image](https://user-images.githubusercontent.com/104761482/198939285-49b3767e-721e-4ab9-9542-ccd1406e4136.png)